loleaflet/src/map/handler/Map.Feedback.js

```diff
@@ -0,0 +1,62 @@
+/* -*- js-indent-level: 8 -*- */
+/*
+ * L.Map.Feedback.
+ */
+
+L.Map.mergeOptions({
+	feedback: true,
+	feedbackTimeout: 2000
+});
+
+L.Map.Feedback = L.Handler.extend({
+
+	addHooks: function () {
+		window.localStorage.setItem('WSDFeedbackEnabled', 'true');
+		this._map.on('docloaded', this.onDocLoaded, this);
+		L.DomEvent.on(window, 'message', this.onMessage, this);
+	},
+
+	removeHooks: function () {
+		L.DomEvent.off(window, 'message', this.onMessage, this);
+	},
+
+	onDocLoaded: function () {
+		setTimeout(L.bind(this.onFeedback, this), this._map.options.feedbackTimeout);
+	},
+
+	onFeedback: function () {
+		if (window.localStorage.getItem('WSDFeedbackEnabled')) {
+			if (this._iframeDialog && this._iframeDialog.hasLoaded())
+				this._iframeDialog.remove();
+
+			this._iframeDialog = L.iframeDialog(window.feebackLocation);
+		}
+	},
+
+	onError: function () {
+		window.localStorage.removeItem('WSDFeedbackEnabled');
+		this._iframeDialog.remove();
+	},
+
+	onMessage: function (e) {
+		var data = e.data;
+
+		if (data == 'show') {
+			this._iframeDialog.show();
+		}
+		else if (data == 'never') {
+			window.localStorage.removeItem('WSDFeedbackEnabled');
+			this._iframeDialog.remove();
+		} else if (data == 'later') {
+			this._iframeDialog.remove();
+			setTimeout(L.bind(this.onFeedback, this), this._map.options.feedbackTimeout);
+		} else if (data == 'submit') {
+			window.localStorage.removeItem('WSDFeedbackEnabled');
+			this._iframeDialog.remove();
+		}
+	}
+});
+
+if (window.isLocalStorageAllowed) {
+	L.Map.addInitHook('addHandler', 'feedback', L.Map.Feedback);
```

Comment:
I think we should add feedback functionality only when `window.feebackLocation` is defined and not empty to not show feedback request for customer builds where it is not wanted

Reply:
I know this file is included only when feedback is enabled but to be sure...